File: frontend/src/api/documents.ts
/**
 * 知识库管理 API(仅 admin 可用,普通用户调用后端返回 403)。
 * 覆盖文档列表、批量上传、删除、分块查看与运营统计,是管理页面的数据来源。
 */
import client from './client'
import type { AdminStats, ChunkItem, DocumentItem, Paged } from '@/types/api'

export const documentApi = {
  // keyword 为文件名模糊搜索:列表页搜索框直接复用该参数,无需单独搜索接口。
  list(page = 1, page_size = 20, keyword?: string) {
    return client.get<Paged<DocumentItem>>('/documents', { params: { page, page_size, keyword } })
  },
  // 批量上传:多个文件打包成一个 FormData,一次请求全部入库(后端逐个解析)。
  upload(files: File[]) {
    const form = new FormData()
    files.forEach((f) => form.append('files', f))
    return client.post<{ uploaded: number }>('/documents/upload', form, {
      timeout: 0, // 大文件上传不限时
    })
  },
  // 删除文档会级联清空其全部向量(后端异步执行),被删除内容将不再被检索到。
  remove(id: string) {
    return client.delete(`/documents/${id}`)
  },
  // 查看文档切分后的知识块:排查"为什么没检索到"时,可对照块内容与搜索词。
  listChunks(id: string, page = 1, page_size = 50) {
    return client.get<Paged<ChunkItem>>(`/documents/${id}/chunks`, { params: { page, page_size } })
  },
  // 运营统计(文档数/知识块数/用户数):展示在管理页顶部的指标卡片。
  stats() {
    return client.get<AdminStats>('/documents/admin/stats')
  },
}

File: backend/app/api/v1/documents.py
"""知识库管理接口(仅 admin):上传 / 列表 / 详情 / 删除 / chunk 预览 / 统计。"""
import asyncio
import logging

from fastapi import APIRouter, File, Query, UploadFile

from app.core.deps import AdminUser, DbDep
from app.models import Document
from app.schemas.document import (
    AdminStats,
    ChunkListOut,
    ChunkOut,
    DocumentListOut,
    DocumentOut,
)
from app.services import document_service, ingestion_service

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/documents", tags=["documents"])


# 注意:admin/stats 必须先于 {doc_id} 注册,避免被路径参数吞掉
@router.get("/admin/stats", response_model=AdminStats)
async def admin_stats(db: DbDep, _admin: AdminUser):
    return await document_service.admin_stats(db)


@router.get("", response_model=DocumentListOut)
async def list_documents(
    db: DbDep,
    _admin: AdminUser,
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    keyword: str | None = None,
):
    items, total = await document_service.list_documents(db, page, page_size, keyword)
    return DocumentListOut(items=items, total=total, page=page, page_size=page_size)


@router.post("/upload")
async def upload_documents(
    db: DbDep,
    admin: AdminUser,
    files: list[UploadFile] = File(...),
):
    """批量上传并启动入库任务;立即返回文档列表,进度通过 GET /documents 轮询。"""
    created: list[Document] = []
    for f in files:
        doc = await document_service.save_upload(f)
        doc.uploaded_by = admin.id
        db.add(doc)
        await db.flush()
        created.append(doc)
    await db.commit()
    # schedule_ingestion 内部已 create_task,此处直接调度
    for doc in created:
        ingestion_service.schedule_ingestion(doc.id)
    return {"uploaded": len(created), "documents": [DocumentOut.model_validate(d) for d in created]}


@router.get("/{doc_id}", response_model=DocumentOut)
async def get_document(doc_id: str, db: DbDep, _admin: AdminUser):
    return await document_service.get_document(db, doc_id)


@router.delete("/{doc_id}")
async def delete_document(doc_id: str, db: DbDep, _admin: AdminUser):
    doc = await document_service.get_document(db, doc_id)
    await document_service.delete_document(db, doc)
    return {"ok": True}


@router.get("/{doc_id}/chunks", response_model=ChunkListOut)
async def list_chunks(
    doc_id: str,
    db: DbDep,
    _admin: AdminUser,
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
):
    items, total = await document_service.list_chunks(db, doc_id, page, page_size)
    return ChunkListOut(items=items, total=total, page=page, page_size=page_size)
